api_tests/src/community.spec.ts

```diff
@@ -180,6 +190,80 @@ test("Remove community", async () => {
   );
 });
 
+test("Report a community", async () => {
+  // Create community on alpha
+  let alphaCommunity = await createCommunity(alpha);
+  expect(alphaCommunity.community_view.community).toBeDefined();
+
+  // Send report from beta
+  let betaCommunity = (
+    await resolveCommunity(beta, alphaCommunity.community_view.community.ap_id)
+  ).community!;
+  let betaReport = (
+    await reportCommunity(beta, betaCommunity.community.id, randomString(10))
+  ).community_report_view.community_report;
+  expect(betaReport).toBeDefined();
+
+  // Report was federated to alpha
+  let alphaReport = (
+    (await waitUntil(
+      () =>
+        listReports(alpha).then(p =>
+          p.reports.find(r => {
+            return checkCommunityReportName(r, betaReport);
+          }),
+        ),
+      res => !!res,
+    ))! as CommunityReportView
+  ).community_report;
+  expect(alphaReport).toBeDefined();
+  expect(alphaReport.resolved).toBe(false);
+  expect(alphaReport.original_community_name).toBe(
+    betaReport.original_community_name,
+  );
+  expect(alphaReport.original_community_title).toBe(
+    betaReport.original_community_title,
+  );
+  expect(alphaReport.original_community_banner).toBe(
+    betaReport.original_community_banner,
+  );
+  expect(alphaReport.original_community_description).toBe(
+    betaReport.original_community_description,
+  );
+  expect(alphaReport.original_community_icon).toBe(
+    betaReport.original_community_icon,
+  );
+  expect(alphaReport.original_community_sidebar).toBe(
+    betaReport.original_community_sidebar,
+  );
+  expect(alphaReport.reason).toBe(betaReport.reason);
+
+  // Resolve report as admin of the community's instance
+  let resolveParams: ResolveCommunityReport = {
+    report_id: alphaReport.id,
+    resolved: true,
+  };
+  let resolve = await alpha.resolveCommunityReport(resolveParams);
+  expect(resolve.community_report_view.community_report.resolved).toBeTruthy();
+
+  // Report should be marked resolved on reporter's instance
+  let resolvedReport = (
+    (await waitUntil(
+      () =>
+        listReports(beta).then(p =>
+          p.reports.find(r => {
+            return (
+              checkCommunityReportName(r, alphaReport) && r.resolver != null
+            );
+          }),
+        ),
+      res => !!res,
+    ))! as CommunityReportView
+  ).community_report;
+  expect(resolvedReport).toBeDefined();
+  expect(resolvedReport.resolved).toBe(true);
```

Comment:
If the community was not removed then the report should stay unresolved, right?

https://github.com/LemmyNet/lemmy/pull/5496/files#diff-59b565f03f82044908e9c09f294e4a28c6a7e32ef9894527ca70cb027ed0943dR377

Reply:
For admins on the report creator's instance, that's close enough to what happens, because it stays in the unresolved reports list. But the updated `resolved` and `resolver` fields from the remote instance are preserved, so local admins can still see it.

Reply:
So the report is still shown despite being resolved? How can admins hide it then without removing the community?

Reply:
Resolving it again will remove it from the list, since the filter checks if the resolver is local.

Reply:
That seems unintuitive. Besides the same reasoning could be applied to local reports for remote posts or comments. Its better to keep them all consistent, and make a separate issue/pull request to handle this case.